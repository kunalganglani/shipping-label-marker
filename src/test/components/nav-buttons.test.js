import React from 'react';
import renderer from 'react-test-renderer';
import NavButtons from '../../core/nav-buttons';
test('NavButtons renders correctly', () => {
  const state = {
    currentStep: 5,
  };
  const props = {
    steps: [1, 2, 3, 4, 5],
  };
  const nextSteps = jest.fn();
  const previousSteps = jest.fn();
  const isCurrentPageValid = () => true;
  const tree = renderer
    .create(<NavButtons
      nextDisabled={isCurrentPageValid ? !isCurrentPageValid() : false}
      current={state.currentStep}
      stepLength={props.steps.length}
      nextSteps={nextSteps}
      previousSteps={previousSteps}
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
