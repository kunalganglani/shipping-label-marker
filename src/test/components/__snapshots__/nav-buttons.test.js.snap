// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavButtons renders correctly 1`] = `
<div
  className="css-1w23u0l"
>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary"
    disabled={false}
    onBlur={[Function]}
    onClick={[MockFunction]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Back
    </span>
  </button>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
    disabled={false}
    onBlur={[Function]}
    onClick={[MockFunction]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Confirm
    </span>
  </button>
</div>
`;
